extension/lib/webcompat.js

```diff
@@ -229,19 +279,129 @@ class WebCompat {
     return targetNode.__compat;
   }
 
-  _getSupportState(browser, version, compatNode, ...terms) {
+  _getCompatSummary(browsers, database, ...terms) {
+    if (!this._hasTerm(database, ...terms)) {
+      return { invalid: true, unsupportedBrowsers: [] };
+    }
+
+    const unsupportedBrowsers = browsers.filter(browser => {
+      const state = this._getSupportState(browser, database, ...terms);
+      return state !== _SUPPORT_STATE.SUPPORTED;
+    });
+    const { deprecated, experimental } = this._getStatus(database, ...terms);
+
+    return {
+      database,
+      terms,
+      deprecated,
+      experimental,
+      unsupportedBrowsers: unsupportedBrowsers.length ? unsupportedBrowsers : [],
```

Comment:
`const unsupportedBrowsers = browsers.filter(...)` will be an empty array if no filter matches were found. So there's no need to explicitly check  for `unsupportedBrowsers.length` here in order to default to an empty array. This could safely become:

```diff
- unsupportedBrowsers: unsupportedBrowsers.length ? unsupportedBrowsers : [],
+ unsupportedBrowsers,
```

Reply:
Indeed! Thank you for finding!